### PR TITLE
Bug 1433244 - allow imageimports to be long runing requests

### PR DESCRIPTION
**Pull request by soltysh**
Fixes https://bugzilla.redhat.com/show_bug.cgi?id=1433244.

### CODE REVIEW

File: pkg/client/imagestreams.go [hunk 1]
```diff
@@ -109,7 +110,7 @@ func (c *imageStreams) UpdateStatus(stream *imageapi.ImageStream) (result *image
 // will be returned if no actual import was requested (the to fields were not set), or an ImageStream if import was requested.
 func (c *imageStreams) Import(isi *imageapi.ImageStreamImport) (*imageapi.ImageStreamImport, error) {
 	result := &imageapi.ImageStreamImport{}
-	if err := c.r.Post().Namespace(c.ns).Resource("imageStreamImports").Body(isi).Do().Into(result); err != nil {
+	if err := c.r.Post().Namespace(c.ns).Resource("imageStreamImports").Timeout(2 * time.Minute).Body(isi).Do().Into(result); err != nil {
```
Comment (soltysh) [inline pkg/client/imagestreams.go]: @smarterclayton this 2m is wrong, we should use `request-timeout` value when specified in client, imho.

Comment (smarterclayton) [inline pkg/client/imagestreams.go]: Isn't request timeout already set into the value we send to the server?  If not, why not?

Comment (soltysh) [inline pkg/client/imagestreams.go]: Yes, request timeout is set and verified inside `WithTimeoutForNonLongRunningRequests` filter, but [resthandler.go](https://github.com/openshift/origin/blob/612dcfb20d42755959895cc19285b429864ddeba/vendor/k8s.io/kubernetes/pkg/apiserver/resthandler.go#L350) has additional timeout which needs to be set here explicitly. Personally, I'd prefer the value from `request-timeout` flag would be applied to this, as well.

**Comment by mfojtik** [comment]
I'm pretty much against adding more logic to handle "wait for" (like --follow, etc). Either we need some solution that will work for all cases where users want to "wait for something to happen" or have some consistent flag (like `--follow`). I would much prefer the first one...

And I think we actually have something like that... `oc observe` can serve as 'wait for tags to be imported'. Maybe we just need to provide more examples on how to use it, or make some derivate of `oc observe` that will explicitely allow users to 'wait for field XYZ is populated or get value XYZ'

**Comment by smarterclayton** [comment]
I'm fine with request-timeout being applied there - that's the effective
request timeout

**Comment by mfojtik** [comment]
as discussed on arch call, i'm fine with this for now.

**Comment by mfojtik** [comment]
[test]

**Comment by mfojtik** [comment]
[test]

**Comment by mfojtik** [comment]
@soltysh 

```
POST /oapi/v1/namespaces/integration/imagestreamimports?timeout=2m0s: (2m0.000878996s) 504
```

this does not seems right?

**Comment by soltysh** [comment]
@smarterclayton ptal if the 2nd commit passing `request-timeout` value all the way down to `Request` object makes sense, if so I'll open an upstream PR for that.

File: vendor/k8s.io/kubernetes/pkg/client/restclient/client.go [hunk 2]
```diff
@@ -222,9 +222,11 @@ func (c *RESTClient) Verb(verb string) *Request {
 	backoff := c.createBackoffMgr()
 
 	if c.Client == nil {
-		return NewRequest(nil, verb, c.base, c.versionedAPIPath, c.contentConfig, c.serializers, backoff, c.Throttle)
+		return NewRequest(nil, verb, c.base, c.versionedAPIPath, c.contentConfig,
+			c.serializers, backoff, c.Throttle, 0)
 	}
-	return NewRequest(c.Client, verb, c.base, c.versionedAPIPath, c.contentConfig, c.serializers, backoff, c.Throttle)
+	return NewRequest(c.Client, verb, c.base, c.versionedAPIPath, c.contentConfig,
```
Comment (smarterclayton) [inline vendor/k8s.io/kubernetes/pkg/client/restclient/client.go]: Timeout should be a helper on Request, vs adding something to the constructor.  There are lots of reasons to set timeouts per request type.

**Comment by smarterclayton** [comment]
We can merge imagestreamimport separately - let's split the other issue out.  There's enough wrinkles we need to deal with there that doing two is better than one.

File: vendor/k8s.io/kubernetes/pkg/client/restclient/client.go (hunk 2 above)
Comment (soltysh) [inline vendor/k8s.io/kubernetes/pkg/client/restclient/client.go]: Spawned https://github.com/openshift/origin/pull/13701

**Comment by soltysh** [comment]
@smarterclayton this is only the fix that allows extended timeout for image imports.

**Comment by smarterclayton** [comment]
LGTM [merge]

**Pull request merged**